src/goto-programs/goto_convert.cpp

```diff
@@ -804,7 +779,7 @@ void goto_convertt::convert_assign(
 
     if(lhs.id()==ID_typecast)
     {
-      assert(lhs.operands().size()==1);
+      INVARIANT(lhs.operands().size() == 1, "typecast takes one operand");
```

Comment:
I believe this should use `to_typecast_expr` in some way (and then not be using `.op0()` below), which would take care of this invariant. Doing so avoids spelling out conditions multiple times across the code base.